### PR TITLE
Add server config option for tz aware timestamps [iso7]

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

On iso7 we try to enable the option by default

Part of https://github.com/inmanta/inmanta-core/issues/6428

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/protocol/common.py [hunk 1]
```diff
@@ -980,10 +981,10 @@ def attach_warnings(code: int, value: Optional[JsonType], warnings: Optional[Lis
     return code, value
 
 
-def json_encode(value: ReturnTypes) -> str:
+def json_encode(value: ReturnTypes, tz_aware: bool = False) -> str:
```
Comment (sanderr) [inline src/inmanta/protocol/common.py:984]: I think it might be better to make this `True` by default (for all methods). Places where you currently don't explicitly set the option (again, for all methods):
- `inmanta.data` -> doesn't really matter here except that if we change it the attribute hash will change but only once. I think this is acceptable but @wouterdb knows best
- `inmanta.agent.handler` -> doesn't matter too much but I think we can safely change it so I think we should
- `inmanta.data.model` -> doesn't matter at all
- `gzipped_json` method 5 lines down from this comment -> if I understand correctly this is what the client uses to send data to the server. I think we can safely change this one
- `inmanta.export` -> doesn't matter too much but again, no reason not to
- `inmanta.resources` -> doesn't matter at all
- `inmanta.protocol.openapi.converter` -> I think it doesn't matter here

Comment (sanderr) [reply src/inmanta/protocol/common.py]: - `inmanta.data`: should've been more clear: I'm talking about the `Resource.make_hash` method here.

File: src/inmanta/server/services/resourceservice.py [hunk 2]
```diff
@@ -748,7 +747,7 @@ async def resource_action_update(
         started: datetime.datetime,
         finished: datetime.datetime,
         status: Optional[Union[const.ResourceState, const.DeprecatedResourceState]],
-        messages: List[Dict[str, Any]],
```
Comment (wouterdb) [inline src/inmanta/server/services/resourceservice.py:751]: This I believe is not possible, we can't just change this part of the external api?

File: tests/test_cli.py [hunk 3]
```diff
@@ -382,19 +383,19 @@ def assert_nr_records_in_output_table(output: str, nr_records: int) -> None:
         finished=datetime.datetime.now(),
         status=ResourceState.failed,
         messages=[
-            {
```
Comment (wouterdb) [inline tests/test_cli.py:385]: this you can not do I think, older agents will still send the data the old way

File: src/inmanta/protocol/common.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/protocol/common.py]: @sanderr I don't see how this will affect the attribute hash?

Comment (sanderr) [reply src/inmanta/protocol/common.py]: Concluded on Slack: shouldn't affect the attribute hash.

**Review by sanderr — CHANGES_REQUESTED** [review]
I was trying to rebase this to be able to see the diff with the iso6 PR but then I realized that we'd actually better merge it in two phases as well: first the iso6+master part, then the master-specific part on top of that. Because otherwise the history becomes a mess and we'll never be able to make sense of what changed exactly if we come back to this PR. Could you make that happen, and request review on the revised PR?

File: src/inmanta/protocol/openapi/converter.py [hunk 4]
```diff
@@ -49,10 +49,10 @@
 from inmanta.types import ReturnTypes
 
 
-def openapi_json_encoder(o: object) -> Union[ReturnTypes, util.JSONSerializable]:
+def openapi_json_encoder(o: object, tz_aware: bool = True) -> Union[ReturnTypes, util.JSONSerializable]:
```
Comment (sanderr) [inline src/inmanta/protocol/openapi/converter.py:52]: I don't really see a need for the parameter here if we never pass it explicitly. I would revert the changes to this file.

File: src/inmanta/protocol/common.py [hunk 5]
```diff
@@ -985,14 +985,14 @@ def attach_warnings(code: int, value: Optional[JsonType], warnings: Optional[Lis
...
-def json_encode(value: ReturnTypes, tz_aware: bool = False) -> str:
+def json_encode(value: ReturnTypes, tz_aware: bool = True) -> str:
     """Our json encode is able to also serialize other types than a dict."""
     # see json_encode in tornado.escape
     return json.dumps(value, default=partial(custom_json_encoder, tz_aware=tz_aware)).replace("</", "<\\/")
 
 
-def gzipped_json(value: JsonType) -> Tuple[bool, Union[bytes, str]]:
-    json_string = json_encode(value)
+def gzipped_json(value: JsonType, tz_aware: bool = True) -> Tuple[bool, Union[bytes, str]]:
```
Comment (sanderr) [inline src/inmanta/protocol/common.py:994]: The changes to this method too seem redundant.

**Review by sanderr — CHANGES_REQUESTED** [review]

**Review by sanderr — APPROVED** [review]
